### PR TITLE
make pack sdk fully localized

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fixes: https://github.com/NuGet/Home/issues/4336

This PR ilmerges all the dependent satellite DLLs for NuGet.Build.Tasks.Pack per culture and adds it to the nupkg.

### CODE REVIEW

**Review by emgarten — APPROVED** [review]

**Review by dtivel — APPROVED** [review]

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj [hunk 1]
```diff
@@ -51,13 +51,25 @@
...
 
-  <Target Name="ILMergeNuGetPack"  Condition="'$(TargetFramework)' != ''">
+  <Target Name="ILMergeNuGetPackResourcesPerCulture" DependsOnTargets="Publish">
+    <ItemGroup>
+      <_PackResourceDlls Include="$(OutputPath)**\$(AssemblyName).resources.dll"/>
+    </ItemGroup>
+    <MSBuild
+      Projects="$(MSBuildProjectFullPath)"
+      Targets="ILMergeNuGetPackResources"
+      Properties="CurrentCulture=%(_PackResourceDlls.RecursiveDir);
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj:67]: do you need a distinct here? Multiple files may have the same parent directory.

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj [hunk 2]
```diff
@@ -73,6 +85,22 @@
...
+    <PropertyGroup>
+      <ILMergeResultDir>$(OutputPath)ilmerge\$(CurrentCulture)</ILMergeResultDir>
+    </PropertyGroup>
+    <MakeDir Directories="$(ILMergeResultDir)"/>
+    <ItemGroup>
+      <ResourceArtifacts Include="$(OutputPath)$(CurrentCulture)NuGet*.resources.dll" Exclude="$(OutputPath)$(CurrentCulture)*Pack.resources.dll" />
+    </ItemGroup>
+    <PropertyGroup>
+      <PathToMergedNuGetPackResource>$(ILMergeResultDir)NuGet.Build.Tasks.Pack.resources.dll</PathToMergedNuGetPackResource>
+      <IlmergeCommand>$(ILMergeExePath) $(OutputPath)$(CurrentCulture)NuGet.Build.Tasks.Pack.resources.dll @(ResourceArtifacts, ' ') /out:$(PathToMergedNuGetPackResource) /lib:$(DotnetSharedSDKDirectory) /lib:$(OutputPath)$(RuntimeIdentifier)\publish /internalize /xmldocs /log:$(ILMergeResultDir)IlMergeLog.txt</IlmergeCommand>
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj:98]: Do we need `/lib:` here? Since we are only merging the resources....

**Review by mishra14 — CHANGES_REQUESTED** [review]
Some clarifications on code and - 
1. What does the package look like?
2. Have we verified this with cli team?

-----------------------------------------------------
Just saw the email, nupkg looks good.

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj (hunk 1 above)
Comment (rohit21agrawal) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj:67]: nope, it only looks for it's own resources file - $(AssemblyName).resources.dll

**Review by mishra14 — APPROVED** [review]

**Pull request merged**